File: algorithms/sort/insertion_sort.py
# insertion sort in python
def insertion_sort(nums):
    for i in range(1, len(nums)):
        current_value = nums[i]
        current_position = i
        
        while current_position > 0 and nums[current_position - 1] > current_value:
            nums[current_position] = nums[current_position - 1]
            current_position = current_position - 1
        nums[current_position] = current_value
    return nums
    
nums = [3, 7, 2, 5, 1, 9, 0]
x = insertion_sort(nums)
print(x)

File: algorithms/sort/InsertionSort.java
// Insertion sort in java
import java.util.Arrays;

class InsertionSort {
    
    void insertionSort(int[] array) {
        
        for (int i = 1; i < array.length; i++) {
            int currentValue = array[i];
            int currentPosition = i;
            
            // compare currentValue with each element on the left of it 
            // until an element smaller than it is found
            while (currentPosition > 0 && currentValue < array[currentPosition - 1]) {
                array[currentPosition] = array[currentPosition - 1];
                currentPosition = currentPosition - 1;
            }
            array[currentPosition] = currentValue;
        }
    }
    public static void main(String[] args) {
        int[] array = {5, 2, -1, 15, -4};
        InsertionSort insertSort = new InsertionSort();
        insertSort.insertionSort(array);
        System.out.println(Arrays.toString(array));
    }
}
